includes/Admin/Admin_AJAX.php

```diff
@@ -108,12 +108,17 @@ public function set_up_environment() {
 			);
 		}
 
-		$checks = filter_input( INPUT_POST, 'checks', FILTER_DEFAULT, FILTER_FORCE_ARRAY );
-		$plugin = filter_input( INPUT_POST, 'plugin', FILTER_SANITIZE_FULL_SPECIAL_CHARS );
+		$categories = filter_input( INPUT_POST, 'categories', FILTER_DEFAULT, FILTER_FORCE_ARRAY );
+		$categories = is_null( $categories ) ? array() : $categories;
+		$checks     = filter_input( INPUT_POST, 'checks', FILTER_DEFAULT, FILTER_FORCE_ARRAY );
+		$checks     = is_null( $checks ) ? array() : $checks;
+		$plugin     = filter_input( INPUT_POST, 'plugin', FILTER_SANITIZE_FULL_SPECIAL_CHARS );
 
 		try {
 			$runner->set_check_slugs( $checks );
 			$runner->set_plugin( $plugin );
+			$runner->filter_checks_by_specific_categories( $categories );
```

Comment:
These changes look like they're happening in the wrong place. The categories are only important to filter the checks to run, which should happen in the `get_checks_to_run()` AJAX request, which is the very first one. Subsequent requests should only receive the actual list of checks, so the categories shouldn't be relevant for those requests anymore.

I think the changes in this method should be reverted, since at this point we should already know the proper list of checks.